superset-frontend/packages/superset-ui-core/test/utils/lruCache.test.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { lruCache } from '@superset-ui/core';
+
+test('initial LRU', () => {
+  expect(lruCache().capacity).toBe(100);
+  expect(lruCache(10).capacity).toBe(10);
+  expect(lruCache(10).size).toBe(0);
+  expect(() => lruCache(0)).toThrow(Error);
+});
+
+test('LRU operation', () => {
+  const cache = lruCache<string>(3);
+  cache.set('1', 'a');
+  cache.set('2', 'b');
+  cache.set('3', 'c');
+  cache.set('4', 'd');
+  expect(cache.size).toBe(3);
+  expect(cache.has('1')).toBeFalsy();
+  expect(cache.get('1')).toBeUndefined();
+  cache.get('2');
+  cache.set('5', 'e');
+  expect(cache.has('2')).toBeTruthy();
+  expect(cache.has('3')).toBeFalsy();
+  // @ts-ignore
+  expect(() => cache.set(0)).toThrow(TypeError);
```

Comment:
```suggestion
  // @ts-expect-error
  expect(() => cache.set('set-key-with-no-value')).toThrow(TypeError);
```

I think `ts-expect-error` is now preferred for such use cases. Also, we can make a small change to indicate why an error is expected.

Reply:
currently, LRU supports to set a key without value, the default fallback value is `undefined`. I totally agree with using node-lru-cache if we have more requirements arise.